File: tests/003.pas
var
  a, b, n, k : integer;

begin
  Readln(n);
  k := 0;
  a := 0;
  while (k = 0) and (a <= n div 3) do
  begin
    b := 0;
    while (k = 0) and (b <= n div 5) do
    begin
      if 3 * a + 5 * b = n then
      begin
        writeln(a, ' ', b);
        k := k + 1;
      end;
      b := b + 1
    end;
    a := a + 1
  end;
  if k = 0 then
    writeln('IMPOSSIBLE');
end.

File: tests/003.py
n = int(input())
k = 0
a = 0
while (k=0)and(a<=n//3):
    b = 0
    while (k=0)and(b<=n//5):
        if 3*a+5*b=n:
            print(a,' ',b)
            k = k+1
        b = b+1
    a = a+1
if k=0:
    print('IMPOSSIBLE')
